docs/_posts/2021-05-19-employing-right-configurations-for-hudi-cleaner.md

```diff
@@ -0,0 +1,77 @@
+ - **KEEP_LATEST_FILE_VERSIONS**: This is a static numeric policy that has the effect of keeping N number of file versions irrespective of time. This policy is use-ful when it is known how many MAX versions of the file does one want to keep at any given time. To achieve the same behaviour as before of preventing long running queries from failing, one should do their calculations based on data patterns. Alternatively, this policy is also useful if a user just wants to maintain 1 latest version of the file.
+
+### Examples
+
+Suppose a user uses the below configs for cleaning:
+
+```java
+hoodie.cleaner.policy=KEEP_LATEST_COMMITS
+hoodie.cleaner.commits.retained=10
+```
+
+Cleaner selects the versions of files to be cleaned by taking care of the following:
+
+ - Latest version of a file should not be cleaned.
+ - The commit times of the last 10 (configured) + 1 commits are determined. One extra commit is included because the time window for retaining commits is essentially equal to the longest query run time. So if the longest query takes 5 hours to finish, and ingestion happens every 30 minutes, you need to retain last 10 commits since 10*30 = 300 (5 hours). At this point of time, the longest query can still be using files written in 11th commit in reverse order.  Now for any file group, only those file slices are scheduled for cleaning which are not savepointed (another Hudi table service) and whose commit time is less than the 11th commit in reverse order.
+
+Suppose a user uses the below configs for cleaning:
+
+```java
+hoodie.cleaner.policy=KEEP_LATEST_FILE_VERSIONS
+hoodie.cleaner.fileversions.retained=2
+```
+
+Cleaner does the following:
+
+ - For any file group, last 2 versions (including any for pending compaction) of file slices are kept and the rest are scheduled for cleaning.
+
+### Configurations
+
+You can find the details about all the possible configurations along with the default values [here](https://hudi.apache.org/docs/configurations.html#compaction-configs).
+
+### Run command
+
+Hudi's cleaner table service can be run as a separate process or along with your data ingestion. As mentioned earlier, it basically cleans up any stale/old files lying around. In case you want to run it along with ingesting data, configs are available which enable you to run it in [parallel or in sync](https://hudi.apache.org/docs/configurations.html#withAsyncClean). You can use the below command for running the cleaner independently:
```

Comment:
I guess we can call out we have hudi-cli capability as well.

Reply:
Thank you for pointing this out as well. Added the commands.

Also once we have cleans commands documented properly here - https://hudi.apache.org/docs/deployment.html#cli
We can add a link to the page in this blog as well.